src/nvim/msgpack_rpc/channel.c

```diff
@@ -321,13 +332,16 @@ static void parse_msgpack(Channel *channel)
       }
       arena_mem_free(arena_finish(&p->arena));
     } else if (p->type == kMessageTypeResponse) {
-      ChannelCallFrame *frame = kv_last(channel->rpc.call_stack);
-      if (p->request_id != frame->request_id) {
+      const char *client_type = rpc_client_type(channel);
+      ChannelCallFrame *frame = strcmp(client_type, "msgpack-rpc") == 0
+        ? find_call_frame(&channel->rpc, p->request_id)
+        : kv_last(channel->rpc.call_stack);
+      if (frame == NULL || p->request_id != frame->request_id) {
         char buf[256];
         snprintf(buf, sizeof(buf),
-                 "ch %" PRIu64 " returned a response with an unknown request "
-                 "id. Ensure the client is properly synchronized",
-                 channel->id);
+                 "ch %" PRIu64 " (%s) returned a response with an unknown request "
```

Comment:
Good instinct! Thank you for enhancing the message.
```suggestion
                 "ch %" PRIu64 " (type=%s) returned a response with an unknown request "
```